src/Microsoft.Azure.SignalR.Management/ServiceManagerBuilder.cs

```diff
@@ -39,16 +54,26 @@ public ServiceManagerBuilder WithCallingAssembly()
         /// <returns>The instance of the <see cref="IServiceManager"/>.</returns>
         public IServiceManager Build()
         {
-            _options.ValidateOptions();
+            if (ServiceProvider != null)
+            {
+                throw new InvalidOperationException($"Mulitple invocation of the method is not allowed.");
```

Comment:
why we have such restriction is because we want UT to test `ServiceProvider`? otherwise `ServiceProvider` can be disposed inside `Build()`?

Should the following case be a valid one?
```cs
var manager1 = builder.WithOptions(o=>o.ConnectionString="one").Build();
var manager2 = builder.WithOptions(o=>o.ConnectionString="another").Build();
```

Reply:
If `ServiceProvider` is disposed, then all the services built by service provider are disposed. We should only dispose `ServiceProvider` only when we don't need `ServiceManager` anymore.

Reply:
Now `ServiceProvider` is disposed from `ServiceManager`